src/libraries/Common/src/Interop/Windows/Kernel32/Interop.GetComputerNameEx.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Kernel32
+    {
+        [DllImport(Libraries.Kernel32, EntryPoint = "GetComputerNameExW", SetLastError = true, CharSet = CharSet.Unicode, ExactSpelling = true)]
+        // The documentation for GetComputerNameEx says type BOOL but returns zero or nonzero rather than 0 or 1.
+        // To avoid hacked bools and potential bad code generation; we compare the result with 0 explicitly.
+        internal static extern unsafe uint GetComputerNameEx(COMPUTER_NAME_FORMAT NameType, char *lpBuffer, ref uint nSize);
```

Comment:
It'd be a bit faster / lighterweight to change the `ref uint` to instead be `uint*`.  The method and call site already need to be "unsafe" to support the `char*`, and by making this a pointer instead of a ref, it means there's no stub required to pin the uint just in case it's part of some object on the heap.
```suggestion
        internal static extern unsafe uint GetComputerNameEx(COMPUTER_NAME_FORMAT NameType, char *lpBuffer, uint* nSize);
```

Reply:
That's a good change.